source/core/options.ts

```diff
@@ -674,11 +674,12 @@ const defaultInternals: Options['_internals'] = {
 			return JSON.parse(response.body as string);
 		},
 		paginate: ({response}) => {
-			if (typeof response.headers.link !== 'string') {
+			const rawLinkHeader = response.headers.link;
+			if (typeof rawLinkHeader !== 'string' || rawLinkHeader === '') {
```

Comment:
Should it also handle whitespace? Just to be completely resilient.

```suggestion
			if (typeof rawLinkHeader !== 'string' || rawLinkHeader.trim() === '') {
```

Reply:
Nice suggestion 👍